src/core/shape_inference/include/pad_shape_inference.hpp

```diff
@@ -152,5 +161,24 @@ void shape_infer(const Pad* op,
 }
 
 }  // namespace v1
+
+namespace v12 {
+template <class TShape>
+std::vector<TShape> shape_infer(const Pad* op,
+                                const std::vector<TShape>& input_shapes,
+                                const std::map<size_t, HostTensorPtr>& constant_data = {}) {
+    return util::shape_infer(op, input_shapes, constant_data);
+}
```

Comment:
is this function needed? I thought only `util::shape_infer` is only needed for implementation of `PadBase`.

Reply:
Actually `std::vector<TShape> shape_infer(const Pad*,...)` was redundant, so I have removed this function. 
But `void op::v1::shape_infer(const Pad*,...)` is currently used by GPU, so at this point I would like to keep it for compatibility.